### PR TITLE
Use the new API for DonationWidget

### PR DESCRIPTION
**Pull request by zner0L**
The API changed significantly but that allowed me to clean up the code for the widget a bit. I moved most of the payment provider specific code to the server (where it should have been in the first place).

**Please merge and deploy datenanfragen/backend#7 first.**

### CODE REVIEW

**Comment by baltpeter** [comment]
We can now kick `donate.datenanfragen.de` from the CSP: https://github.com/datenanfragen/website/blob/6d580c6cadda3ebf5d5e49163af6181023b5d04b/_headers#L9

File: src/Components/DonationWidget.js [hunk 1]
```diff
@@ -277,113 +273,92 @@ export default class DonationWidget extends preact.Component {
...
+                        custom: this.state.donation_reference
                     },
+                    '_top'
+                );
+                break;
+            case 'mollie':
+            case 'creditcard':
+            case 'cryptocurrency':
+            default:
+                // eslint-disable-next-line no-case-declarations
```
[the remark this answers is not in the document]
Comment (zner0L) [reply src/Components/DonationWidget.js:327]: Nope, just stupidity.

File: src/Components/DonationWidget.js [hunk 2]
```diff
@@ -277,113 +273,92 @@ export default class DonationWidget extends preact.Component {
...
-                    success_url: `${BASE_URL}donate?donation_reference=${this.state.donation_reference}`,
-                    cancel_url: `${BASE_URL}donate`,
-                    order_id: this.state.donation_reference
-                }
-            };
+                    reference: this.state.donation_reference,
+                    redirect_base: BASE_URL
+                };
+                this.setState({ ongoing_request: true });
+                console.log(donation);
```
[the remark this answers is not in the document]
Comment (zner0L) [reply src/Components/DonationWidget.js:336]: We should have a pre-commit test for that really.

[the remark this answers is not in the document]
Comment (baltpeter) [reply src/Components/DonationWidget.js:336]: I thought we did, at least at some point. I am explicitly disabling that check in the error handler:

https://github.com/datenanfragen/website/blob/6d580c6cadda3ebf5d5e49163af6181023b5d04b/src/error-handler.js#L2

[the remark this answers is not in the document]
Comment (baltpeter) [reply src/Components/DonationWidget.js:336]: Oh, I know why. We have enabled the Node preset in ESLint which (understandably) disables this rule. I'll submit a PR to explicitly enable it.

**Pull request merged**